app/services/usps_exporter.rb

```diff
@@ -0,0 +1,54 @@
+class UspsExporter
+  OTP_MAX_VALID_DAYS = 30
+
+  def initialize(csv_file_path)
+    @csv_file_path = csv_file_path
+  end
+
+  def run
+    CSV.open(csv_file_path, 'wb', col_sep: '|') do |csv|
+      make_csv(csv)
+    end
+    clear_entries
+  end
+
+  private
+
+  attr_reader :csv_file_path, :entry_ids
+
+  def entries
+    entries = UspsConfirmation.all
+    @entry_ids = entries.map(&:id)
+    entries.map(&:decrypted_entry)
+  end
+
+  def clear_entries
+    UspsConfirmation.where(id: entry_ids).destroy_all
+  end
+
+  def make_csv(csv)
+    entries.each do |entry|
+      csv << make_entry_row(entry)
+    end
+  end
+
+  # rubocop:disable MethodLength, AbcSize
+  def make_entry_row(entry)
+    now = Time.zone.now
+    due = now + OTP_MAX_VALID_DAYS.days
+    current_date = now.strftime('%B %e')
+    due_date = due.strftime('%B %e')
```

Comment:
not sure what you were suggesting with putting `due_date` in a different method, but I did inline the strftime calls.